### PR TITLE
feat: improve rest openapi response codes

### PR DESCRIPTION
**Pull request by tiagolobocastro**
1. feat: improve errors codes and resp 200 schema
    Add new RestClusterError for operations where 404 does not make sense.
    Add a new Unit type which does not expose a schema on the openapi and
    will look better on the swagger-ui since we're just returning OK with
    no associated data.

2. feat: expose openapi error responses
    Expose all openapi error responses and an associated error schema for
    more detailed information about the error.

3. chore: update paperclip and lock async_trait
    Point paperclip to our own repo with the latest fixes until a new
    release is published to crates-io.
    Lock async_trait because of an imcompatibility with the tracing crate.
    Make the mayastor derivation depend on the $rev so that nix is aware of
    this and knows when it needs to rebuild it.
    Set panics to abort on dev builds.

4. refactor(auth): add new authorization error type
    Add error type to the authentication module which we can now plug into
    the RestError and will be displayed returned to the requester as part
    of the response body.

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

File: control-plane/rest/src/versions/v0.rs [hunk 1]
```diff
@@ -541,55 +541,113 @@ impl ActixRestClient {
 }
 
 /// Rest Error
+#[api_v2_errors(
+    code = 400,
+    description = "Request Timeout",
+    code = 500,
+    description = "Internal Server Error",
+    code = 500,
+    description = "Internal Server Error",
```
Comment (paulyoong) [inline control-plane/rest/src/versions/v0.rs:550]: Duplicated

File: control-plane/rest/src/versions/v0.rs [hunk 2]
```diff
@@ -541,55 +541,113 @@ impl ActixRestClient {
...
+    code = 503,
+    description = "Service Unavailable",
+    code = 416,
+    description = "Range Not satisfiable",
+    code = 501,
+    description = "Not Implemented",
+    code = 503,
+    description = "Service Unavailable",
+    code = 401,
+    description = "Unauthorized",
```
Comment (paulyoong) [inline control-plane/rest/src/versions/v0.rs:574]: Duplicated on line 559

**Review by paulyoong — APPROVED** [review]

File: control-plane/rest/src/versions/v0.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/rest/src/versions/v0.rs:550]: dohhh

File: control-plane/rest/src/versions/v0.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/rest/src/versions/v0.rs:574]: dohhh

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**